### PR TITLE
Add function documentation, clean up function conventions for consistency

### PR DESCRIPTION
**Pull request by austinvhuang**
- [ ] Add documentation to untyped Functional module

### CODE REVIEW

File: hasktorch/src/Torch/Functional.hs [hunk 1]
```diff
@@ -91,175 +91,407 @@ instance Castable Reduction Int64 where
...
+-- | mean
+-- Returns the mean value of all elements in the input tensor.
+mean 
+    :: Tensor -- ^ input
+    -> Tensor -- ^ output
 mean t = unsafePerformIO $ (cast1 ATen.mean_t) t
 
-std :: Tensor -> Tensor
+-- | std
+-- Returns the standard-deviation of all elements in the input tensor.
```
Comment (adamwespiser) [inline hasktorch/src/Torch/Functional.hs]: "standard deviation"

Comment (austinvhuang) [reply hasktorch/src/Torch/Functional.hs]: Thanks, fixed

File: hasktorch/src/Torch/Functional.hs [hunk 2]
```diff
@@ -455,10 +766,17 @@ conv2d' weight bias stride padding input =
...
-solve :: Tensor -> Tensor -> (Tensor,Tensor)
+solve 
+    :: Tensor 
+    -> Tensor 
+    -> (Tensor,Tensor)
 solve b a = unsafePerformIO $ (cast2 ATen.solve_tt) b a
 
-cholesky_inverse :: Tensor -> Tri -> Tensor
+-- | Solves a linear system of equations with a positive semidefinite matrix to be inverted given its Cholesky factor matrix uu .
+cholesky_inverse
```
Comment (championballer) [inline hasktorch/src/Torch/Functional.hs]: @austinvhuang Should this also be camelCase? Similar to the choleskySolve change made in this PR?

Comment (austinvhuang) [reply hasktorch/src/Torch/Functional.hs]: Yes it should be, just haven't gotten to all the bits to cleanup in this PR.

Basically the current informal convention is basically - code generated (such as the Functional/Internal module and everything in libtorch-ffi) is snake case, user facing is camelCase in line with Haskell convention, but there are still bits here and there to clean up (BCE is another one).

Feel free to add PR to fix other user-facing instances.

**Comment by austinvhuang** [comment]
I'm going to go ahead and merge this so others can add cleanup PRs without overlapping with changes here.

File: hasktorch/src/Torch/Functional.hs (hunk 2 above)
Comment (championballer) [reply hasktorch/src/Torch/Functional.hs]: Gotcha. Thanks. :)

**Pull request merged**